### PR TITLE
fix glob-to-regexp d.ts

### PR DESCRIPTION
**Pull request by AlexTugarev**
Signed-off-by: Alex Tugarev <alex.tugarev@typefox.io>

### CODE REVIEW

**Comment by AlexTugarev** [comment]
The issue is, that `DocumentFilter`s like `{ pattern: ".*"}` cannot be used, as the type definition refers to an ES6 module, but `glob-to-regexp` is not a such. Also, the function is only called if only `pattern` is provided.

File: typings/glob-to-regexp/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 declare module "glob-to-regexp" {
```
Comment (akosyakov) [inline typings/glob-to-regexp/index.d.ts]: Do we need to delcare module then at all? https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html looks like we should be able to declare a function and export it

File: src/languages.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /* --------------------------------------------------------------------------------------------
- * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Copyright (c) 2018 TypeFox GmbH (http://www.typefox.io). All rights reserved.
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
-import globToRegExp from "glob-to-regexp";
+import * as globToRegExp from 'glob-to-regexp';
```
Comment (akosyakov) [inline src/languages.ts]: `import globToRegExp = require('glob-to-regexp')` should be used

File: typings/glob-to-regexp/index.d.ts (hunk 1 above)
Comment (AlexTugarev) [reply typings/glob-to-regexp/index.d.ts]: module declaration is needed if `import ... from "module"` syntax should stay, cf. https://github.com/Microsoft/TypeScript/issues/5073.

File: src/languages.ts (hunk 2 above)
Comment (AlexTugarev) [reply src/languages.ts]: should I change it?

Comment (AlexTugarev) [reply src/languages.ts]: 👍

File: typings/glob-to-regexp/index.d.ts (hunk 1 above)
Comment (akosyakov) [reply typings/glob-to-regexp/index.d.ts]: if we use `import x = require('module')` then it should not be necessary

Comment (AlexTugarev) [reply typings/glob-to-regexp/index.d.ts]: well, it's still required.

> [ts]
Could not find a declaration file for module 'glob-to-regexp'. '/Users/tugarev/Projects/monaco-languageclient/node_modules/glob-to-regexp/index.js' implicitly has an 'any' type.
  Try `npm install @types/glob-to-regexp` if it exists or add a new declaration (.d.ts) file containing `declare module 'glob-to-regexp';`

Comment (AlexTugarev) [reply typings/glob-to-regexp/index.d.ts]: I double-checked that, it's required if types should be imported, right? module-function.d.ts seems to be misleading.

**Comment by akosyakov** [comment]
@AlexTugarev I've sent you an invitation, please accept it and merge the PR

**Pull request merged**